src/inmanta/data/__init__.py

```diff
@@ -155,19 +155,39 @@ def __init__(self, message: str, *args) -> None:
         self.message = message
 
 
-class DatabaseOrder(metaclass=ABCMeta):
+class DatabaseOrder:
     """Represents an ordering for database queries"""
 
+    valid_sort_columns: Dict[str, Union[Type[datetime.datetime], Type[int], Type[str]]] = {}
+
+    @classmethod
+    def validator_dataclass(cls) -> Type["BaseDocument"]:
+        """ The class used for checking whether the ordering is valid for a table"""
+        return BaseDocument
+
+    @classmethod
+    def parse_from_string(
+        cls,
+        sort: str,
+    ) -> "DatabaseOrder":
+        valid_sort_pattern: Pattern[str] = re.compile(f"^({'|'.join(cls.valid_sort_columns)})\\.(asc|desc)$", re.IGNORECASE)
+        match = valid_sort_pattern.match(sort)
+        if match and len(match.groups()) == 2:
+            order_by_column = match.groups()[0].lower()
+            validated_order_by_column, validated_order = cls.validator_dataclass()._validate_order_strict(
```

Comment:
This validation checks the name, but not the type. 

I wonder if it would not be more interesting to make a testcase that goes over every subclass of DatabaseOrder and verifies if `valid_sort_columns` is always valid wrt to the `BaseDocument`

Reply:
In that case we could derive the types from the `__fields` of the specific `BaseDocument` (and `valid_sort_columns` could be only a list of the names), but since we also have cases where we sort on nested columns (for the `ResourceLog`s), that would also require specific tweaking only for that use case.
For the time being I think it's an acceptable compromise that if a developer mixes up the types here, the tests would fail.

In the longer term, for the refactoring epic, I agree that this information should come from one source only.